controller/aks-cluster-config-handler.go

```
@@ -0,0 +1,686 @@
+		upstreamSpec.NetworkDNSServiceIP = to.String(networkProfile.DNSServiceIP)
+		upstreamSpec.NetworkDockerBridgeCIDR = to.String(networkProfile.DockerBridgeCidr)
+		upstreamSpec.NetworkServiceCIDR = to.String(networkProfile.ServiceCidr)
+		upstreamSpec.NetworkPolicy = string(networkProfile.NetworkPolicy)
+		upstreamSpec.NetworkPodCIDR = to.String(networkProfile.PodCidr)
+		upstreamSpec.LoadBalancerSku = string(networkProfile.LoadBalancerSku)
+	}
+
+	// set linux account profile
+	linuxProfile := clusterState.LinuxProfile
+	if linuxProfile != nil {
+		upstreamSpec.LinuxAdminUsername = to.String(linuxProfile.AdminUsername)
+		sshKeys := *linuxProfile.SSH.PublicKeys
+		upstreamSpec.LinuxSSHPublicKeyContents = to.String(sshKeys[0].KeyData)
+	}
+	return upstreamSpec, err
+}
+
+func updateUpstreamClusterState(ctx context.Context, config *aksv1.AKSClusterConfig, upstreamSpec *aksv1.AKSClusterConfigSpec) (*aksv1.AKSClusterConfig, error) {
+	// check kubernetes version for update
+
+	if config.Spec.Tags == nil {
+		config.Spec.Tags = map[string]string{}
+	}
+	config.Spec.Tags["displayName"] = config.Spec.DisplayName
+	if reflect.DeepEqual(config.Spec.Tags, upstreamSpec.Tags) {
+		logrus.Debugf("[AKS] updateUpstreamClusterState check config are similar, exit \n")
+		return config, nil
+	}
+
+	azureAuthorizer, _ := newClientAuthorizer(config.Spec)
+	resourceGroupsClient, _ := newResourceGroupsClient(azureAuthorizer, config.Spec)
+
+	exists, _ := checkResourceGroup(ctx, resourceGroupsClient, config.Spec)
+	if !exists {
+		logrus.Infof("Resource group %s does not exist, creating", config.Spec.ResourceGroup)
+		status, err := createResourceGroup(ctx, resourceGroupsClient, config.Spec)
+		if status {
+			logrus.Infof("Resource grup [%s] updated successfully", config.Name)
+		}
+		if err != nil {
+			fmt.Errorf("Error during creating resource group %v", err)
+		}
+	}
+
+	resourceClusterClient, _ := newClustersClient(azureAuthorizer, config.Spec)
```

Are we ignoring this error?

---

this was fixed

---

actually the err part was fixed but not the nil check on the client return